utils/bitfield/src/rleplus/writer.rs

```diff
@@ -1,6 +1,9 @@
 // Copyright 2019-2022 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+/// The maximum varint that can be serialized on 9 bytes.
+const MAX_VARINT: u64 = 2u64.pow(63)-1;
```

Comment:
What do you think about using `usize` here? Unless we're supporting 32-bit systems it should be fine, plus we would get rid of some casts.

Reply:
Good question. We indeed want 32-bit systems support like `wasm32`. Protocol Labs had to port our code to explicitly use `u64` (there are many commits, the one for Bitfield is [here](https://github.com/filecoin-project/ref-fvm/commit/95e1f33a4433cb55d4ad0ff75b013e4e8b539b30)). We will probably use some of their crates instead of ours (I'm commiting here but this PR will have to be done on the `ref-fvm` repo too to keep things in sync in the meantime).

Even if we don't want to support say 32-bit system I think it's a good practice to use properly size type (makes the code more readable and correct).

Reply:
Thanks for the explanation! In this case, should we rather do an opposite cast which wouldn't be lossy? Given that potentially (on a 32-bit machine) `MAX_VARINT as usize` would get truncated to 32 bits (`u32::MAX`). In the end the behaviour would still be the same - on a 32 bit machine the condition will always pass regardless.